eZ/Bundle/EzPublishCoreBundle/Command/CleanupVersionsCommand.php

```diff
@@ -0,0 +1,266 @@
+
+    const VERSION_DRAFT = 'draft';
+    const VERSION_ARCHIVED = 'archived';
+    const VERSION_PUBLISHED = 'published';
+    const VERSION_ALL = 'all';
+
+    const VERSION_STATUS = [
+        self::VERSION_DRAFT => VersionInfo::STATUS_DRAFT,
+        self::VERSION_ARCHIVED => VersionInfo::STATUS_ARCHIVED,
+        self::VERSION_PUBLISHED => VersionInfo::STATUS_PUBLISHED,
+    ];
+
+    /**
+     * @var \eZ\Publish\API\Repository\Repository
+     */
+    private $repository;
+
+    /**
+     * @var \eZ\Publish\API\Repository\UserService
+     */
+    private $userService;
+
+    /**
+     * @var \eZ\Publish\API\Repository\ContentService
+     */
+    private $contentService;
+
+    /**
+     * @var \eZ\Publish\API\Repository\PermissionResolver
+     */
+    private $permissionResolver;
+
+    /**
+     * @var \eZ\Bundle\EzPublishCoreBundle\ApiLoader\RepositoryConfigurationProvider
+     */
+    private $repositoryConfigurationProvider;
+
+    /**
+     * @var \Doctrine\DBAL\Driver\Connection
+     */
+    private $connection;
+
+    public function __construct(
+        Repository $repository,
+        RepositoryConfigurationProvider $repositoryConfigurationProvider,
+        Connection $connection
+    ) {
+        $this->repository = $repository;
+        $this->repositoryConfigurationProvider = $repositoryConfigurationProvider;
+        $this->connection = $connection;
+
+        parent::__construct();
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        parent::initialize($input, $output);
+
+        $this->userService = $this->repository->getUserService();
+        $this->contentService = $this->repository->getContentService();
+        $this->permissionResolver = $this->repository->getPermissionResolver();
+
+        $this->permissionResolver->setCurrentUserReference(
+            $this->userService->loadUserByLogin($input->getOption('user'))
+        );
+    }
+
+    protected function configure()
+    {
+        $config = $this->repositoryConfigurationProvider->getRepositoryConfig();
+
+        $this
+            ->setName('ezplatform:content:cleanup-versions')
+            ->setDescription('Remove unwanted content versions. It keeps published version untouched. By default, it keeps also the last archived/draft version.')
+            ->addOption(
+                'status',
+                't',
+                InputOption::VALUE_OPTIONAL,
+                sprintf(
+                    "Select which version types should be removed: '%s', '%s', '%s'.",
+                    self::VERSION_DRAFT,
+                    self::VERSION_ARCHIVED,
+                    self::VERSION_ALL
+                ),
+                self::VERSION_ALL
+            )
+            ->addOption(
+                'keep',
+                'k',
+                InputOption::VALUE_OPTIONAL,
+                "Sets number of the most recent versions (both drafts and archived) which won't be removed.",
```

Comment:
Not sure having 1 value for the sum of both archived versions and drafts makes a lot of sense here.
I generally operate on the basis of "keep the last 4 archived versions", "keep all drafts", "keep no drafts".
I see no reason for anyone ever wanting to keep only the 4 most recent drafts, or, even weirder, the 4 most recent drafts-or-archives

Reply:
ps: my bad, reading the code it seems that the number of versions to keep is not a sum, which makes a lot more sense. Maybe the help text could be improved a bit to explain this more clearly ?